apps/showcase/src/assets/trainings/sdk/steps/model-extension/solution/core/flight/flight.reviver.ts

```diff
@@ -0,0 +1,19 @@
+import type { Flight } from '../../base/flight/flight';
+import type { reviveFlight } from '../../base/flight/flight.reviver';
+import type { FlightCoreIfy } from './flight';
+
+/**
+ * Generate reviver for Flight core model
+ *
+ * @param baseRevive
+ */
+export function reviveFlightFactory<R extends typeof reviveFlight>(baseRevive: R) {
+  const reviver = <T extends Flight = Flight>(data: any, dictionaries?: any) => {
+    const revivedData = baseRevive<FlightCoreIfy<T>>(data, dictionaries);
+    /* Set the value of your new fields here */
+    revivedData.id = 'sampleIdValue';
+    return revivedData;
```

Comment:
It could be nice to show the id somewhere in a ui?

Reply:
But they won't get the autocompletion :/ - at least the compiler should complain

Reply:
the revived object is now shown in the preview